File: swea/D2/2005.파스칼의 삼각형/2005.cpp
#include <iostream>
using namespace std;

int T;

int main(){
    cin >> T;

    for(int t = 1; t <= T; t++){
        int n;
        cin >> n;

        int arr[10][10] = {0};

        for(int i = 0; i < n; i++){
            arr[i][0] = 1;
        }

        for(int i = 1; i <= n; i++){
            for(int j = 1; j <= i; j++){
                arr[i][j] = arr[i-1][j] + arr[i-1][j-1];
            }
        }

        cout << '#' << t << endl;
        for(int i = 0; i < n; i++){
            for(int j = 0; j <= i; j++){
                cout << arr[i][j] << ' ';
            }
            cout << endl;
        }
    }

    return 0;
}

File: swea/D2/2005.파스칼의 삼각형/2005.py
T = int(input())
for tc in range(1, T+1):
    n = int(input())

    arr = [[0] * n for _ in range(n)]
    arr[0][0] = 1

    for i in range(1, n):
        for j in range(n):
            arr[i][j] = arr[i-1][j-1] + arr[i-1][j]
    
    print(f"#{tc}")
    for i in range(n):
        for j in range(n):
            if arr[i][j] != 0:
                print(arr[i][j], end=" ")
        print()